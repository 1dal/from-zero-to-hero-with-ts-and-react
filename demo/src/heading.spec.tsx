import React from 'react';
import { shallow } from 'enzyme';

import { Heading, Props } from './heading';

describe(`<Heading/>`, () => {
  it(`should render`, () => {
    const wrapper = shallow(<Heading message="Hello"/>)

    expect(wrapper.contains(<h1>Hello</h1>)).toBeTruthy()
  })

  it(`should assert via snapshot`, () => {

    const wrapper = shallow<Props>(<Heading message="React is great!"/>)

    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({message:'Foo Bar'})

    expect(wrapper).toMatchSnapshot()
  })
})
