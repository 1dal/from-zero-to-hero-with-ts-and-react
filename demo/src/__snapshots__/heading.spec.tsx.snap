// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading/> should assert via snapshot 1`] = `
<h1>
  React is great!
</h1>
`;

exports[`<Heading/> should assert via snapshot 2`] = `
<h1>
  Foo Bar
</h1>
`;
